app/services/notification/process.rb

```diff
@@ -0,0 +1,99 @@
+require 'enumerate_it'
+
+class Notification::Process
+  class TransactionStatus < ::EnumerateIt::Base
+    associate_values(
+      authorized:     1,
+      started:        2,
+      printed_boleto: 3,
+      finished:       4,
+      canceled:       5,
+      process:        6,
+      written_back:   7,
+      refunded:       9
+    )
+  end
+
+  def initialize(params)
+    @params  = params
+  end
+
+  def self.call(params)
+    new(params).call
+  end
+
+  def call
+    process_moip_message unless payment.nil?
+  end
+
+  private
+  attr_accessor :params
+
+  def payment
+    @payment ||= find_payment
+  end
+
+  def find_payment
+    payment = PaymentEngines.find_payment(key: params[:id_transacao])
```

Comment:
What about the case when payment is `nil`? Wouldn't be better to throw an error?